Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.ts

```diff
@@ -0,0 +1,323 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+
+type Item = { UniqueIdentifier: string, DisplayName: string }
+
+class BocAutoCompleteReferenceValue
+{
+
+  private static _itemBackUp: Nullable<Item>;
```

Comment:
Those are instance fields

Reply:
Does that work? C# only calls static members and methods and does not save an instance of the class to operate on it as far as I'm aware

Reply:
That's not correct. The initialize function got all the parameters and did a ton of closures and thus passed the values along when intantiating the closure. Now, a bunch of statics is populated with information that's specfic to individual calls of Initialize. Best case, it works but the data is just invalid after each invocation and the final invocations leaves said invalid data behind. Worst case: something gets captured and a field is lazy evaluated and then the subsequent calls have wrong state.

Either way: the expressed semantics would be wrong.